<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <!-- Begin Jekyll SEO tag v2.7.1 -->
<title>BLASTing Illumina reads in FASTQ format | SequenceServer</title>
<meta name="generator" content="Jekyll v3.9.1" />
<meta property="og:title" content="BLASTing Illumina reads in FASTQ format" />
<meta name="author" content="Yannick" />
<meta property="og:locale" content="en_US" />
<meta name="description" content="BLAST is powerful, but not necessarily apprpriate for analysis of FASTQ files" />
<meta property="og:description" content="BLAST is powerful, but not necessarily apprpriate for analysis of FASTQ files" />
<link rel="canonical" href="https://sequenceserver.com/blog/blast_illumina_reads_in_fastq_format/" />
<meta property="og:url" content="https://sequenceserver.com/blog/blast_illumina_reads_in_fastq_format/" />
<meta property="og:site_name" content="SequenceServer" />
<meta property="og:type" content="article" />
<meta property="article:published_time" content="2021-11-01T13:03:32+00:00" />
<meta name="twitter:card" content="summary" />
<meta property="twitter:title" content="BLASTing Illumina reads in FASTQ format" />
<script type="application/ld+json">
{"@type":"BlogPosting","headline":"BLASTing Illumina reads in FASTQ format","dateModified":"2021-11-01T13:03:32+00:00","datePublished":"2021-11-01T13:03:32+00:00","mainEntityOfPage":{"@type":"WebPage","@id":"https://sequenceserver.com/blog/blast_illumina_reads_in_fastq_format/"},"author":{"@type":"Person","name":"Yannick"},"description":"BLAST is powerful, but not necessarily apprpriate for analysis of FASTQ files","url":"https://sequenceserver.com/blog/blast_illumina_reads_in_fastq_format/","@context":"https://schema.org"}</script>
<!-- End Jekyll SEO tag -->
<meta name="author" content="Anurag Priyam">
  <meta name="author" content="Yannick Wurm">
  <meta name="keywords"
    content="run blast on mac, local NCBI BLAST, custom BLAST, local BLAST server, BLAST web frontend, local BLAST, BLAST web interface, galaxy, viroblast">
  <!-- Support HTML5 elements and media queries on IE8 and less. -->
  <!--[if lt IE 9]>
      <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.2/html5shiv.min.js"></script>
    <![endif]-->

  <!-- Bootstrap and FontAwesome CSS -->
  <link rel="stylesheet" type="text/css" href="/css/googlefont-source-sans-pro.css">
  <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.0.2/dist/css/bootstrap.min.css" rel="stylesheet"
    integrity="sha384-EVSTQN3/azprG1Anm3QDgpJLIm9Nao0Yz1ztcQTwFspd3yD65VohhpuuCOmLASjC" crossorigin="anonymous">
  <link rel="stylesheet" type="text/css" href="/css/fontawesome-4.3.0.css">
  <link rel="stylesheet" type="text/css" href="/css/pygments-github.css">
  <link rel="stylesheet" type="text/css" href="/css/seqserv.css">

  <!-- Google Analytics -->
  <script>
    (function (i, s, o, g, r, a, m) {
      i['GoogleAnalyticsObject'] = r; i[r] = i[r] || function () {
        (i[r].q = i[r].q || []).push(arguments)
      }, i[r].l = 1 * new Date(); a = s.createElement(o),
        m = s.getElementsByTagName(o)[0]; a.async = 1; a.src = g; m.parentNode.insertBefore(a, m)
    })(window, document, 'script', '//www.google-analytics.com/analytics.js', 'ga');

    ga('create', 'UA-471841-7', 'sequenceserver.com');
    ga('send', 'pageview');
  </script>
</head>

<body>
  <nav class="navbar navbar-expand-xl navbar-light fixed-top">
    <div class="container">
      <div class="brand-container">
        <a class="navbar-brand navbar-left" href="/"><img class="logo" src="/img/logos/SequenceServer_logo.png"
							  alt="sequenceserver_logo">&nbsp;<span>1.0.14</span></a>
	<a class="navbar-brand-made-by" href="//wurmlab.com">
          made by Wurmlab @ Queen Mary University of London
        </a>
      </div>
      <button class="navbar-toggler" type="button" data-bs-toggle="collapse"
        data-bs-target="#navbarTogglerDemo02" aria-controls="navbarTogglerDemo02" aria-expanded="false"
        aria-label="Toggle navigation">
        <span class="navbar-toggler-icon"></span>
      </button>
      <div class="collapse navbar-collapse" id="navbarTogglerDemo02">
        <ul class="navbar-nav me-auto mb-2 mb-lg-0">
          <li class="nav-item">
            <a class="nav-link" href="/#installation">
              Installation
            </a>
          </li>
          <li class="nav-item">
            <a class="nav-link" href="/doc/">
              Advanced configuration
            </a>
          </li>
          <li class="nav-item">
            <a class="nav-link" href="/cloud/" style="font-weight: bold; background-color:#ffcccc;">
              Hosted cloud service
            </a>
          </li>
          <li class="nav-item">
            <a class="nav-link" href="/#license">
              License
            </a>
          </li>
          <li class="nav-item">
            <a class="nav-link" href="/#support">
              Support
            </a>
          </li>
          <li class="nav-item">
            <a class="nav-link" href="/#users">
              Users
            </a>
          </li>
          <li class="nav-item">
            <a class="visible-lg nav-link" href="https://antgenomes.sequenceserver.com" title="Will open in new window"
              target="_blank" rel="noopener">
              <i class="fa fa-external-link" aria-hidden="true"></i>
              Demo
            </a>
          </li>
          <li class="nav-item">
            <a class="visible-lg nav-link" href="https://github.com/wurmlab/sequenceserver"
              title="Will open GitHub in new window" target="_blank" rel="noopener">
              <i class="fa fa-external-link" aria-hidden="true"></i>
              Code
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
  <div class="container">
    <div class="row row-cols-1 row-cols-sm-2 row-cols-md-3 row-cols-lg-4 g-4 py-5 stats-banner justify-content-center" role="banner">
    <div class="">
        <div>
            <p><a href="/#users"><i class="fa fa-quote-left" aria-hidden="true"></i> 130+ citations</a></p>
        </div>
    </div>
    <div class="col d-flex align-items-start">
        <div>
            <p><i class="fa fa-users" aria-hidden="true"></i> 10,000+ users</p>
        </div>
    </div>
    <div class="col d-flex align-items-start">
        <div>
            <p><a href="https://rubygems.org/gems/sequenceserver" target="_blank" rel="noreferrer"><i class="fa fa-download" aria-hidden="true"></i> 90,000+ downloads</a></p>
        </div>
    </div>
    <div class="col d-flex align-items-start">
        <div>
            <p><a href="https://github.com/wurmlab/sequenceserver" target="_blank" rel="noreferrer"><i class="fa fa-star" aria-hidden="true"></i> 217 stars on GitHub</a></p>
        </div>
    </div>
</div>

    <h1 id="blasting-illumina-reads-in-fastq-format">BLASTing Illumina reads in FASTQ format</h1>

<p>BLAST uses FASTA format for queries and for database creation. So the BLAST algorithm doesn’t directly understand FASTQ format. This is in part because BLAST was created long before the FASTQ format was created, and because FASTQ files are <em>typically</em> inappropriate for BLAST analysis.</p>

<p>FASTQ files typically are the result of Illumina or Nanopore sequencing. FASTQ files thus include a lot of redundancy (many reads from the same subset of the genome or transcriptome, or from a particualr amplicon. When this is the case:</p>

<ul>
  <li>BLAST analysis will be <strong>slow</strong> because the algorithm needs to search through a much larger set of sequences than if redundancy had been removed. Indeed, FASTQs are typically huge files containing tens to hundreds of millions of read sequences.</li>
  <li>If results are found, they are likely to include a lot of redundancy (many similar reads obtaining high scores). This <strong>makes interpretation difficult</strong>.</li>
  <li>Having a particularly large set of redundant sequences to search through also <strong>reduces BLAST’s ability to identify sequence similarities</strong>. This is because BLAST’s detection power depends on the size of the query and database. Indeed, the e-value of a particular alignment is lower if your database or search are larger. (E-value is grossly equivalent  to “the number of times I’d find this match by chance if the database were made up of random nucleotides”).</li>
</ul>

<h2 id="most-of-the-time-if-you-want-ot-blast-a-fastq-file-youre-probably-not-using-the-best-approach">Most of the time if you want ot BLAST a FASTQ file, you’re probably not using the best approach</h2>

<p>It is likely that you want to first reduce redundancy in your dataset. The most biologically relevant way is often to perform whole genome or transcriptome assembly of your raw reads prior to BLASTing them.</p>

<p>If you do want to work with the raw reads, BLAST often isn’t the best way to perform analysis.</p>

<h2 id="but-what-if-i-really-do-need-to-run-blast-on-fastq-files">But what if I really do need to run BLAST on FASTQ files?</h2>

<p>With the aforementioned caveats that it often is inappropriate to BLAST raw reads, gaining biological insight sometimes does depend on it.</p>

<p>The easiest way of converting FASTQ to FASTA format is to use something like the following <code class="language-plaintext highlighter-rouge">seqtk</code> command:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>seqtk seq -A input.fq &gt; output.fasta
</code></pre></div></div>

<p>There are many other ways - I recommend using a tried and tested tool rather than creating your own thing by creatively using grep or perl.</p>

<div class="container">
  <div class="row justify-content-center">
  <div class="alert alert-info" style="max-width:75%">
   <p>By leveraging cloud computing and publication-ready graphics, SequenceServer Cloud makes it easy to perform sequence search results and to interpret them. <a href="https://sequenceserver.com/cloud/">Learn more</a></p>
   <p style="text-align:center"><a href="https://sequenceserver.com/cloud/"><img src="/img/logos/SequenceServer_logo.png" alt="Sequence Search with SequenceServer" width="200pt" /></a></p>
    </div>
  </div>
</div>

<!--
## General considerations for design of biological data visualization

Each of the visualizations has helped us in our interpretation of BLAST results. For each visualization, we tried to respect

We
-->

  </div>

  <footer style="background-color: white">
    <hr>
    <div class="container" style="text-align: center; padding: 20px 0;">
      <p>Please cite:
        <a href="/paper/">
	  Sequenceserver: a modern graphical user interface for custom BLAST databases. (2019) Molecular Biology &amp; Evolution</a>
        </a>
      </p>
      <p>
        SequenceServer is &copy; 5bases Ltd (2012-2016) and authors. Parts of SequenceServer are &copy; Universit&eacute; de Lausanne (2011) and &copy; Queen Mary University of London (2012-). The SequenceServer Cloud service is provided by <a href="http://pragmaticgenomics.com">Pragmatic Genomics Limited</a>. 
      </p>
    </div>
  </footer>

  <!-- jQuery and Bootstrap JS -->
  <script src="https://code.jquery.com/jquery-3.6.0.min.js"
    integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script>
  <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.0.2/dist/js/bootstrap.bundle.min.js"
    integrity="sha384-MrcW6ZMFYlzcLA8Nl+NtUVF0sA7MsXsP1UyJoMp4YLEuNSfAP+JcXn/tWtIaxVXM"
    crossorigin="anonymous"></script>
  <script>
    $(function () {
      var navMain = $(".navbar-collapse");
      navMain.on("click", "a:not([data-toggle])", null, function () {
        navMain.collapse('hide');
      });
    });
  </script>
</body>

</html>
